float64[] angles
float64[] gradients
float64 intensity
